src/base/bittorrent/sessionimpl.cpp

```diff
@@ -5741,6 +5747,12 @@ void SessionImpl::handleExternalIPAlert(const lt::external_ip_alert *p)
     }
 }
 
+void SessionImpl::handleSessionErrorAlert(const lt::session_error_alert *p) const
+{
+    LogMsg(tr("Session encountered a serious error. Reason: \"%1\"")
+        .arg(QString::fromLocal8Bit(p->error.message().c_str())), Log::CRITICAL);
```

Comment:
Why not use `alert::message()` instead?
```suggestion
        .arg(QString::fromStdString(p->message())), Log::CRITICAL);
```
(in both cases)

Reply:
IIRC on Windows and using Chinese locale, using `fromStdString` and the message will become garbage. Only using `fromLocal8Bit` will get the correct message. Also I suspect this is because the message comes from boost and boost wasn't using utf-8 encoding but rather the OS default.

Reply:
Never mind, I was talking about something else.
It is done.